src/Options.js

```diff
@@ -250,10 +257,38 @@ export class Options {
 };
 
 
-// A distinguish instance shared internally via module
-//
-export var options = new Options();
+class ParseOptions {
+  constructor(options) {
+    this.proxiedOptions_ = options;
 
+    Object.keys(featureOptions).forEach((name) => {
```

Comment:
Can you move this out of the constructor? Instead, set up the getters on the prototype.